src/steps/os/linux.rs

```diff
@@ -234,6 +237,42 @@ fn upgrade_redhat(ctx: &ExecutionContext) -> Result<()> {
     Ok(())
 }
 
+fn upgrade_nobara(ctx: &ExecutionContext) -> Result<()> {
+    let sudo = require_option(ctx.sudo().as_ref(), REQUIRE_SUDO.to_string())?;
+    let pkg_manager = which("dnf").unwrap_or_else(|| Path::new("yum").to_path_buf());
```

Comment:
If Nobara wants its users to use the `dnf` command, should we still fall back to `yum`?